File: Java/String-1/hasBad.java

public class string1 {
	
	// given a string, return true if "bad" appears starting at index 0 or 1 in the string
	// the string may be any length, including 0
	public boolean hasBad(String str) {
		
		boolean b = false;
		
		if (str.length() >= 3 && str.substring(0, 3).equals("bad"))
			b = true;
		else if (str.length() >= 4 && str.substring(1, 4).equals("bad"))
			b = true;
		
		return b;
		
	}

	public static void main(String[] args) {
		
		string1 o = new string1();
		
		System.out.println(o.hasBad("badxx"));
		System.out.println(o.hasBad("xbadxx"));
		System.out.println(o.hasBad("xxbadxx"));
	
	}

}

File: Python/String-1/has_bad.py

#
#
def has_bad(str):
    b = False

    if len(str) >= 3 and str[:3] == 'bad':
        b = True
    if len(str) >= 4 and str[1:4] == 'bad':
        b = True

    return b

print(has_bad('badxx'))
print(has_bad('xbadxx'))
print(has_bad('xxbadxx'))
